Migrate Bubble Sort and add unit tests

alxmjo opened the pull request:
- Move implementation to header file
- Add sorting cpp file (which can be reused for other sorting algos)
- Add include guard to utils.hpp
- Update readme
- Update build script
- Remove old file

@faheel This is my basic template for sorting algorithm migration, so I will leave this open for you until you have a look at it.

alxmjo committed:
Migrate Bubble Sort and add unit tests

- Move implementation to header file
- Add sorting cpp file (which can be reused for other sorting algos)
- Add include guard to utils.hpp
- Update readme
- Update build script
- Remove old file

faheel commented on C++/test/algorithm/sorting/sorting.cpp:
```diff
@@ -0,0 +1,82 @@
+	v.reserve(vectorSize);
+
+	for (int i = 0; i < (int) vectorSize; i++) {
+		v.push_back(generate_random_int(-32767, 32767));
```
I think we should increase the range to -2^31 to 2^31 - 1. Or it would be even better to use [`std::numeric_limits`](https://en.cppreference.com/w/cpp/types/numeric_limits).

alxmjo replied:
Noted. I'll use `std::numeric_limits<int>::min()`.

alxmjo replied:
Was working on `counting_sort.hpp` today and realized that this wide range of integers is causing the algorithm to run extremely slowly ([more info](https://stackoverflow.com/questions/25001800/why-we-can-not-apply-counting-sort-to-general-arrays)). I'm going to use the following going forward:

`v.push_back(generate_random_int(std::numeric_limits<short>::min(), std::numeric_limits<short>::max()));`

This prevents the use of magic numbers while still giving a reasonably large range of positive and negative values to sort.

faheel replied:
Makes sense :+1:

faheel commented on C++/test/algorithm/sorting/sorting.cpp:
```diff
@@ -0,0 +1,82 @@
+int generate_random_int(int min, int max) {
+	std::default_random_engine generator(std::random_device{}());
+	std::uniform_int_distribution<> intRange(min, max);
+	return intRange(generator);
```
`intRange` should be renamed `int_range` to maintain naming consistency.

alxmjo replied:
Forgot to add that if you have other suggestions I'm more than happy to follow suit.

faheel commented on C++/test/algorithm/sorting/sorting.cpp:
```diff
@@ -0,0 +1,82 @@
+/*
+	generate_unsorted_vector
+	------------------------
+	Generates a vector with a random size (up to 100) and fills with random integers.
```
Based on the code, the comment should say `up to 1000` (instead of 100). Also, I think the following will be better:

```
Creates a vector of random size (up to 1000) and populates it with random integers.
```

It would also be better if the function accepted an argument `max_size` and had 1000 as the default value for it if none was provided. This would make it more flexible, and therefore reusable in even more test case scenarios.

alxmjo committed:
Implement suggested and other changes

- Allowing generate_unsorted_vector to take argument of max_size
- Set default value for max_size
- Variable renaming
- Use of std::numeric_limits to define integer range
- Updating comments
- Replacing magic number with const value for TIMES_TO_RUN

alxmjo commented:
Thanks for the careful review. I've implemented the suggested changes.
